sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/context/AzureGlobalPropertiesAutoConfiguration.java

```diff
@@ -48,13 +64,38 @@ public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata,
                     () -> Binder.get(this.environment)
                                 .bindOrCreate(AzureGlobalProperties.PREFIX,
                                     AzureGlobalProperties.class));
-                if (ClassUtils.isPresent(AZURE_AUTHENTICATION_TEMPLATE_CLASS_NAME, null)) {
+                if (evaluateNonBinderCondition(registry)) {
                     definitionBuilder.addDependsOn(SPRING_TOKEN_CREDENTIAL_PROVIDER_CONTEXT_BEAN_NAME);
                 }
                 registry.registerBeanDefinition(AZURE_GLOBAL_PROPERTY_BEAN_NAME, definitionBuilder.getBeanDefinition());
             }
         }
 
+        private boolean evaluateNonBinderCondition(BeanDefinitionRegistry registry) {
+            if (ClassUtils.isPresent(AZURE_AUTHENTICATION_TEMPLATE_CLASS_NAME, this.classLoader)) {
+                if (registry instanceof DefaultListableBeanFactory beanFactory) {
+                    if (this.environment instanceof StandardEnvironment standardEnvironment) {
+                        MutablePropertySources propertySources = standardEnvironment.getPropertySources();
+                        PropertySource<?> firstPropertySource = propertySources.iterator().next();
+                        if (firstPropertySource instanceof MapPropertySource mapPropertySource
+                            && mapPropertySource.getSource() instanceof HashMap<String, Object> maybeBinderProperties) {
+                            if (maybeBinderProperties.isEmpty() && beanFactory.getParentBeanFactory() != null) {
+                                return false;
+                            }
+                        }
+
+                        if (beanFactory.containsBean("outerContext")
+                            && beanFactory.getBean("outerContext").getClass() == AnnotationConfigApplicationContext.class) {
+                            return false;
+                        }
+
+                        return !firstPropertySource.getName().equals("defaultBinderFactoryProperties")
+                            && propertySources.stream().anyMatch(src -> src.getName().equals("configurationProperties"));
```

Comment:
For these strings: `outerContext`, `defaultBinderFactoryProperties`, `configurationProperties`.
1. Is it possible that these strings change in the future?
2. How about using a variable name instead of string. If the variable exists, use it directly, if it not exist, we can define on in current file.

Reply:
Can we add unit test for this logic?